lib/interactor/context.rb

```diff
@@ -121,10 +121,22 @@ def failure?
     #
     # Raises Interactor::Failure initialized with the Interactor::Context.
     def fail!(context = {})
-      error_context = Context.new
-      context.each { |key, value| error_context[key.to_sym] = value }
-      error_context.failure = @failure = true
-      raise Failure, error_context
+      context.each { |key, value| self[key.to_sym] = value }
+
+      redaction_message = 'REDACTED'
+      self.each_pair do |name, value|
+        if value.is_a?(Hash)
+          if value.has_key?(:password)
+            self[name.to_sym][:password] = redaction_message
```

Comment:
Minor: no `to_sym` needed here. `OpenStruct` stores everything as a symbol under the hood, and makes everything accessible by either `String` or `Symbol` keys. 

```ruby
# Values can be written as String or Symbol
a = OpenStruct.new      # => #<OpenStruct>
a[:foo] = 'one'         # => "one"
a['foo'] = 'two'        # => "two"

# Values can be accessed as either String, Symbol,
# or `.` notation
[a.foo, a['foo'], a[:foo]]  # => ["two", "two", "two"]
```
```